### PR TITLE
dsg-1779: past applications public portal

### PR DESCRIPTION
**Pull request by derekdiep42**
# Basics

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [x] Styling and layout changes
- [ ] Code style changes (formatting, local variables)
- [ ] Refactoring (no functional changes)
- [ ] Build or CI related changes
- [ ] Tests
- [ ] Other... Please describe:

## What changes does this PR introduce?

We now introduce an entirely new public portal loading technique. Essentially, we make a call for the first page (20 applications). if total applications is more, then we have a load more button that loads the second page. We keep all transactions in a subject, and keep appending onto the subject if we make more api calls using the service. This is done for both past applications and active applications.

## Screenshot of the affected changes

Video of load more and past applications:

https://github.com/Nuvalence/dsgov-web/assets/56942479/1566fb2f-a2e7-41d4-933d-1891c3196f1b

Past applications container:

https://github.com/Nuvalence/dsgov-web/assets/56942479/a5e80d86-eef4-46ff-a7ee-7ae266c66fea

## Checklist

- [ ] I have updated the documentation accordingly
- [x] I have confirmed all acceptance criteria defined by the story have been met
- [x] I have confirmed that Sonar Qube findings for new code are reviewed and fixed, or will be whitelisted

### CODE REVIEW

File: libs/public/feature/dashboard/src/lib/components/dashboard/dashboard.component.ts [hunk 1]
```diff
@@ -57,6 +72,14 @@ export class DashboardComponent {
     },
   ];
 
+  public togglePastApplications() {
```
Comment (thechechogomez) [inline libs/public/feature/dashboard/src/lib/components/dashboard/dashboard.component.ts:75]: is this redundant? Don't see where this is being used. Same with line 38 `public showPastApplications = true;`.

File: libs/public/feature/dashboard/src/lib/components/dashboard/dashboard.component.ts [hunk 2]
```diff
@@ -25,15 +27,28 @@ import { DashboardService } from '../../services';
...
   templateUrl: './dashboard.component.html',
 })
 export class DashboardComponent {
+  public showPastApplications = true;
+  public pastApplicationsDisplayLimit = 50;
+  public pastApplicationPanel: INuverialPanel[] = [{ expanded: true, panelTitle: 'Past Applications' }];
   public profile$: Observable<UserModel | null> = this._profileService.getProfile$();
-  public transactions$: Observable<TransactionModel[]> = this._dashboardService.getTransactions$();
+  public transactions$: Observable<{ pastApplications: TransactionModel[]; activeTransactions: TransactionModel[] }> = this._dashboardService
+    .getTransactions$()
```
Comment (thechechogomez) [inline libs/public/feature/dashboard/src/lib/components/dashboard/dashboard.component.ts:43]: Seems `getTransactions` is paginated and we are only asking for the first 10 transactions. If I start creating random transactions, the completed transactions get lost.

**Review by thechechogomez — COMMENTED** [review]
Good job. I liked the template outlet. Just a few comments

File: libs/public/feature/dashboard/src/lib/components/dashboard/dashboard.component.ts (hunk 2 above)
Comment (derekdiep42) [reply libs/public/feature/dashboard/src/lib/components/dashboard/dashboard.component.ts:42]: @thechechogomez Nice catch. I think for now, until we have a plan for how to load more active transactions, I think I'm going to set the request for 200 transactions so we guarantee a fetch for everything.

**Review by davidyoung77 — APPROVED** [review]

File: libs/public/feature/dashboard/src/lib/components/dashboard/dashboard.component.html [hunk 3]
```diff
@@ -12,53 +12,75 @@ <h1>Welcome<span *ngIf="profile$ | async as profile">{{profile.displayName ? ',
...
+  <ng-container *ngIf="transactions.activeTransactions.length; else emptyTransactionsState">
+    <div class="transaction-container" *ngFor="let transaction of transactions.activeTransactions; trackBy:trackByFn">
+      <ng-container *ngTemplateOutlet="transactionTemplate; context: {transaction: transaction}"></ng-container>
+    </div>
+  </ng-container>
+  <section class="seperator"><hr /></section>
+  <ng-container>
+    <nuverial-accordion class="expansion-panel" [panelList]="pastApplicationPanel">
+      <ng-template #body let-panel>
+        <ng-container *ngIf="transactions.pastApplications.length; else emptyTransactionsState">
```
Comment (thechechogomez) [inline libs/public/feature/dashboard/src/lib/components/dashboard/dashboard.component.html:24]: @derekdiep42 One more thing. If you check the comments on [this](https://dol-ewf.atlassian.net/browse/DSG-1818) story, we don't need to show the accordion or the empty state message when there are no past applications.

**Review by thechechogomez — COMMENTED** [review]
Just one small change

**Review by davidyoung77 — APPROVED** [review]

File: libs/public/feature/dashboard/src/lib/components/dashboard/dashboard.component.ts [hunk 4]
```diff
@@ -25,23 +28,56 @@ import { DashboardService } from '../../services';
...
     NuverialSelectComponent,
     NuverialSelectorButtonDropdownComponent,
+    NuverialAccordionComponent,
   ],
   selector: 'dsg-dashboard',
   standalone: true,
   styleUrls: ['./dashboard.component.scss'],
   templateUrl: './dashboard.component.html',
 })
 export class DashboardComponent {
```
Comment (thechechogomez) [inline libs/public/feature/dashboard/src/lib/components/dashboard/dashboard.component.ts:38]: https://github.com/Nuvalence/dsgov-web/assets/60438413/a563ab1d-f6a2-4a7f-a6f2-09c8c9f504a7

I'm getting this error where there are duplicates if I navigate to a transaction and navigate back to the dashboard

Comment (derekdiep42) [reply libs/public/feature/dashboard/src/lib/components/dashboard/dashboard.component.ts]: Fixed! We cleaned up the service now. Good catch!

**Review by thechechogomez — APPROVED** [review]
Looks good, just a small question.

**Review by thechechogomez — APPROVED** [review]

**Pull request merged**